### ISSUE TITLE
background-image 

### ISSUE THREAD

**Issue by zhangchao828** [cssnano/cssnano#812]
input：
background: url('https://ss0.bdstatic.com/70cFuHSh_Q1YnxGkpoWK1HF6hhy/it/u=508387608,2848974022&fm=26&gp=0.jpg?imageView2/1/w/750/h/1334');

output：
background: url('https://ss0.bdstatic.com/70cFuHSh_Q1YnxGkpoWK1HF6hhy/it/u=508387608,2848974022&fm=26&gp=0.jpg?imageView2%2F1%2Fw%2F750%2Fh%2F1334');

/->%2F

**Comment by anikethsaha** [comment]
```css
background: url("https://ss0.bdstatic.com/70cFuHSh_Q1YnxGkpoWK1HF6hhy/it/u=508387608,2848974022&fm=26&gp=0.jpg?imageView2/1/w/750/h/1334");
```

⬇️

```css
background: url(
  https://ss0.bdstatic.com/70cFuHSh_Q1YnxGkpoWK1HF6hhy/it/u=508387608,
  2848974022&fm=26&gp=0.jpg?imageView2%2F1%2Fw%2F750%2Fh%2F1334=
);
```

This looks safe.
Whats the issue !?

@evilebottnawi I guess its good to close now

**Comment by evilebottnawi** [comment]
/cc @anikethsaha 
It is bug, https://tools.ietf.org/html/rfc3986#section-3.4, `/` and `?` allowed in URL, we should fix it

**Comment by anikethsaha** [comment]
yeah but whats wrong if we are transforming them to safe codes like `%` ?

**Comment by evilebottnawi** [comment]
@anikethsaha 

> yeah but whats wrong if we are transforming them to safe codes like % ?

There may be very different servers settings so better keep them as is. Also it is less css

**Comment by evilebottnawi** [comment]
Should be easy to fix, you can try to fix it

**Comment by anikethsaha** [comment]
> Should be easy to fix, you can try to fix it

yeah we need to just leave it as it is right?...no need of any transform.

**Comment by evilebottnawi** [comment]
> yeah we need to just leave it as it is right?...no need of any transform.

yes

**Comment by zhangchao828** [comment]
> @anikethsaha
> 
> > yeah but whats wrong if we are transforming them to safe codes like % ?
> 
> There may be very different servers settings so better keep them as is. Also it is less css

yes

**Comment by taoliujun** [comment]
Still happen at cssnano `4.1.11` because  `sortQueryParameters` default value is not `false`.
Fixed it after set config file, e.g:
```javascript
// cssnano.config.js

module.exports = {
  preset: [
    'default',
    {
      normalizeUrl: {
        sortQueryParameters: false,
      },
    },
  ],
};
```

**Comment by alexander-akait** [comment]
@taoliujun update cssnano to v5

**Comment by taoliujun** [comment]
> @taoliujun update cssnano to v5

Thks for your tip, but cssnano is not in my package.json, it is dependencies by optimize-css-assets-webpack-plugin@^5. 
 I can not upgrade it cross chained dependencies webpack@^4.

**Comment by alexander-akait** [comment]
Unfortunately there is no way to fix it, consider to update webpack to v5 and use https://github.com/webpack-contrib/css-minimizer-webpack-plugin
